File: src/data_models/process_model/start_callback_type.ts
/**
 * Determines when the process engine will resolve after a new process instance/correlation was started.
 */
export enum StartCallbackType {
    /**
     * The process engine will resolve immediately after the process instance was started.
     */
    CallbackOnProcessInstanceCreated = 1,
    /**
     * The process engine will resolve immediately after the first end event was reached.
     */
    CallbackOnProcessInstanceFinished = 2,
    /**
     * The process engine will resolve immediately after a specific end event was reached.
     */
    CallbackOnEndEventReached = 3,
}

File: dotnet/src/data_models/process_model/StartCallbackType.cs
namespace ProcessEngine.ConsumerAPI.Contracts.DataModel
{
    /// <summary>
    /// Determines when the ProcessEngine will finish the returned Task after a
    /// new ProcessInstance was started.
    /// </summary>
    public enum StartCallbackType : byte
    {
        /// <summary>
        /// The ProcessEngine will finish the Task immediately after the
        /// ProcessInstance was started.
        /// </summary>
        CallbackOnProcessInstanceCreated = 1,

        /// <summary>
        /// The ProcessEngine will finish the Task after the first EndEvent was
        /// reached.
        /// </summary>
        CallbackOnProcessInstanceFinished = 2,

        /// <summary>
        /// The ProcessEngine will finish the Task after a specific EndEvent was
        /// reached.
        /// </summary>
        CallbackOnEndEventReached = 3,
    }
}
